### PR TITLE
feat(oss-challenge): Add OSS-Challenge Chart

### PR DESCRIPTION
**Pull request by trowik**

### CODE REVIEW

File: charts/osschallenge/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+sources:
+  - https://github.com/adfinis-sygroup/osschallenge
+  - https://github.com/adfinis-sygroup/helm-charts/tree/master/charts/osschallenge
+maintainers:
+  - name: adfinis-sygroup
+    email: support@adfinis.com
+    url: https://adfinis.com
+dependencies:
+  - name: postgresql
+    version: ~9.1.2
```
Comment (hairmare) [inline charts/osschallenge/Chart.yaml:20]:
```suggestion
    version: ~10.1.0
```

File: charts/osschallenge/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,80 @@
+
```
Comment (hairmare) [inline charts/osschallenge/templates/_helpers.tpl:1]:
```suggestion
```

File: charts/osschallenge/values.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    # osschallenge.readinessProbe.timeoutSeconds -- Number of seconds after which the probe times out
+    timeoutSeconds: 5
+    # osschallenge.readinessProbe.failureThreshold -- Number of tries to perform the probe
+    failureThreshold: 6
+    # osschallenge.readinessProbe.successThreshold -- Minimum consecutive successes for the probe to be considered successful after having failed
+    successThreshold: 1
+  settings:
+    # django settings for osschallenge
+    # osschallenge.settings.emailHost -- Connection string of SMTP server to send mails
+    emailHost: smtp.example.ch
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:70]: Always use a reserved domain in examples/docs.
```suggestion
    emailHost: smtp.example.com
```

File: charts/osschallenge/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v2
+name: osschallenge
+description: Chart for OSS-Challenge application
+type: application
+version: 0.1.0
+appVersion: edge
+keywords:
+  - osschallenge
+  - postgresql
+home: https://github.com/adfinis-sygroup/osschallenge
```
Comment (hairmare) [inline charts/osschallenge/Chart.yaml:10]:
```suggestion
home: https://osschallenge.com
```

**Review by hairmare — CHANGES_REQUESTED** [review]
Here's some initial draft review:

* [ ] update helm-docs to make the linter pass

I'm not quite sure why the deploy test is failing but I'd guess fixing the linter stuff and the other things I commented might help chart-testing write some meaningful output.

File: charts/osschallenge/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+sources:
+  - https://github.com/adfinis-sygroup/osschallenge
+  - https://github.com/adfinis-sygroup/helm-charts/tree/master/charts/osschallenge
+maintainers:
+  - name: adfinis-sygroup
+    email: support@adfinis.com
+    url: https://adfinis.com
+dependencies:
+  - name: postgresql
+    version: ~10.1.0
```
Comment (hairmare) [inline charts/osschallenge/Chart.yaml:20]:
```suggestion
    version: ~10.2.0
```

File: charts/osschallenge/templates/configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ include "osschallenge.fullname" . }}
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+data:
+  EMAIL_HOST: {{ .Values.osschallenge.settings.emailHost | quote }}
+  EMAIL_HOST_PORT: {{ .Values.osschallenge.settings.emailPort | quote }}
+  EMAIL_HOST_USER: {{ .Values.osschallenge.settings.emailUser | quote }}
+  EMAIL_HOST_PASSWORD: {{ .Values.osschallenge.settings.emailPassword | quote }}
```
Comment (hairmare) [inline charts/osschallenge/templates/configmap.yaml:11]: This should not be in a ConfigMap, it should be in a Secret.

File: charts/osschallenge/values.yaml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+postgresql:
+  # postgresql.enabled -- Enable PostgreSQL for persistence
+  enabled: true
+  # postgresql.postgresqlDatabase -- PostgreSQL database name
+  postgresqlDatabase: osschallenge
+  # postgresql.postgresqlUsername -- PostgreSQL user name
+  postgresqlUsername: osschallenge
+  # postgresql.postgresqlPassword -- Set a password for PostgreSQL
+  postgresqlPassword: ~
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:106]: I can't find where this is used. Shouldn't it be in the Secret as well?

File: charts/osschallenge/values.yaml [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  # postgresql.postgresqlPort-- PostgreSQL database port
+  postgresqlPort: 5432
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:108]: Why specify the default port here? Are there any reason for needing to choose a port on a regular basis?

File: charts/osschallenge/values.yaml [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    # osschallenge.settings.emailPort -- Port for the SMTP server
+    emailPort: 587
+    # osschallenge.settings.emailFrom -- Default email address to use for various responses
+    emailFrom: osschallenge@oss-challenge.ch
+    # osschallenge.settings.emailUser -- Username for the SMTP server
+    emailUser: example@example.com
+    # osschallenge.settings.emailPassword -- Password for the SMTP server
+    emailPassword: ~
+    # osschallenge.settings.siteUrl -- URL for OSS-Challenge
+    siteUrl: oss-chalenge.ch
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:80]:
```suggestion
    siteUrl: oss-challenge.ch
```

**Review by hairmare — CHANGES_REQUESTED** [review]

File: charts/osschallenge/templates/secret-backend.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+type: Opaque
+data:
+  {{ if .Values.osschallenge.service.secret }}
+  DJANGO_SECRET_KEY: {{ .Values.osschallenge.settings.secret | b64enc | quote }}
+  {{ else }}
+  DJANGO_SECRET_KEY: {{ randAlphaNum 32 | b64enc | quote }}
+  {{ end }}
+  EMAIL_HOST: {{ .Values.osschallenge.settings.emailHost | quote }}
+  EMAIL_HOST_PORT: {{ .Values.osschallenge.settings.emailPort | quote }}
+  EMAIL_HOST_USER: {{ .Values.osschallenge.settings.emailUser | quote }}
```
Comment (hairmare) [inline charts/osschallenge/templates/secret-backend.yaml:16]: These belong into the ConfigMap and not the Secret... Only secrets should be stored in Secret resources.

File: charts/osschallenge/ci/default-values.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+postgresql:
+  postgresqlPassword: s3cr3t
+  postgresqlPort: 5432
```
Comment (hairmare) [inline charts/osschallenge/ci/default-values.yaml:3]:
```suggestion
```

File: charts/osschallenge/values.yaml [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    # osschallenge.settings.emailPort -- Port for the SMTP server
+    emailPort: 587
+    # osschallenge.settings.emailFrom -- Default email address to use for various responses
+    emailFrom: osschallenge@oss-challenge.ch
+    # osschallenge.settings.emailUser -- Username for the SMTP server
+    emailUser: example@example.com
+    # osschallenge.settings.emailPassword -- Password for the SMTP server
+    emailPassword: ~
+    # osschallenge.settings.siteUrl -- URL for OSS-Challenge
+    siteUrl: oss-challenge.ch
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:80]: This should either be a real URL or be called `stiteHostname` or similar.
```suggestion
    siteUrl: https://oss-challenge.ch
```
```suggestion
    siteHostname: oss-challenge.ch
```

File: charts/osschallenge/values.yaml [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+  # postgresql.enabled -- Enable PostgreSQL for persistence
+  enabled: true
+  # postgresql.postgresqlDatabase -- PostgreSQL database name
+  postgresqlDatabase: osschallenge
+  # postgresql.postgresqlUsername -- PostgreSQL user name
+  postgresqlUsername: osschallenge
+  # postgresql.postgresqlPassword -- Set a password for PostgreSQL
+  # postgresqlPassword: ~
+  # postgresql.postgresqlPort-- PostgreSQL database port
+  # postgresqlPort: 5432
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:108]:
```suggestion
```

File: charts/osschallenge/templates/secret-backend.yaml [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+type: Opaque
+data:
+  {{ if .Values.osschallenge.service.secret }}
+  DJANGO_SECRET_KEY: {{ .Values.osschallenge.settings.secret | b64enc | quote }}
+  {{ else }}
+  DJANGO_SECRET_KEY: {{ randAlphaNum 32 | b64enc | quote }}
+  {{ end }}
+  EMAIL_HOST_PASSWORD: {{ .Values.osschallenge.settings.emailPassword | quote }}
```
Comment (hairmare) [inline charts/osschallenge/templates/secret-backend.yaml:14]:
```suggestion
  EMAIL_HOST_PASSWORD: {{ .Values.osschallenge.settings.emailPassword | b64enc | quote }}
```

File: charts/osschallenge/values.yaml (hunk 7 above)
Comment (hairmare) [reply charts/osschallenge/values.yaml]: found the `{{- include "osschallenge.djangoDatabasePassword" . | nindent 12 }}` lines

File: charts/osschallenge/values.yaml (hunk 12 above)
Comment (hairmare) [reply charts/osschallenge/values.yaml]: Did you re-test oss-challenge after changing this?

Comment (trowik) [reply charts/osschallenge/values.yaml]: I know that the URL is currently in use by on old oss-challenge vm, if that's what you meant.

**Comment by eyenx** [comment]
@trowik pls rebase this as ct lint will be included as of now.

**Comment by trowik** [comment]
> @trowik pls rebase this as ct lint will be included as of now.

done @eyenx

File: charts/osschallenge/templates/service.yaml [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+kind: Service
+metadata:
+  name: {{ include "osschallenge.fullname" . }}
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.osschallenge.service.type }}
+  ports:
+    - port: {{ .Values.osschallenge.service.externalPort }}
+      targetPort: {{ .Values.osschallenge.service.internalPort }}
```
[the remark this answers is not in the document]
Comment (trowik) [reply charts/osschallenge/templates/service.yaml:11]: done as well

**Review by hairmare — APPROVED** [review]

**Pull request merged**